src/vs/workbench/browser/workbench.ts

```diff
@@ -174,6 +176,12 @@ export class Workbench extends Layout {
 
 				// Restore
 				this.restore(lifecycleService);
+
+				// Paste
+				const handlePaste = (event: ClipboardEvent) => {
```

Comment:
Multiple feedback here:
* this is somewhat of a layer break because the generic workbench should not be knowing about a command from the files contribution layer
* I am a bit nervous about running this on a global `paste` event on `window`, is it possible to scope the `paste` listener to the file explorer only and move it into the explorer?

Reply:
Great feedback! I made some changes:

- The `paste` event listener has been moved to `explorerView.ts`. 
- Since for some reason no `paste` events are being fired when the `paste` event listener is on the tree element, it seems the `paste` event listener would need to remain on `window`.  
- It's important that the `paste` event listener on window doesn't interfere with any other parts: The `files.executePaste` command is *only* executed when the explorer is focused.

What do you think? Thanks..!